aptos-move/framework/aptos-framework/sources/governed_gas_pool.move

```diff
@@ -170,6 +172,27 @@ module aptos_framework::governed_gas_pool {
         coin::balance<CoinType>(pool_address)
     }
 
+    /// Withdraws coins from the governed gas pool.
+    ///
+    /// This function allows friend modules to withdraw a specified amount of a given
+    /// `CoinType` from the governed gas pool. It uses the internal signer of the
+    /// governed gas pool to authorize the withdrawal.
+    ///
+    /// @param amount The amount of coins to withdraw from the pool.
+    /// @return A `Coin<CoinType>` resource containing the withdrawn amount.
+    public(friend) fun withdraw_from_pool<CoinType>(
+        amount: u64
+    ): Coin<CoinType> acquires GovernedGasPool {
+        let s = governed_gas_signer(); // uses the private signer function
+        coin::withdraw<CoinType>(&s, amount)
```

Comment:
can we have a dedicated event for withdraw from GGP for accounting purpose

Reply:
Add an event to GGP is a little more complicated than expected. Creating the event handler in the unit test need to have the sender account created, and the framework account is not for most of the test that use the staking.
I've tested, and I get the MISSING_DATA (code 4008) error when the handler is created.

To solve this, you need to add this line in all test that fails: aptos_framework::account::create_account_for_test(@aptos_framework);

I've tested using account::new_event_handle<_>() and event::new_event_handle<_>() , the account must be created to create the handler.

What do you think, I add the create_account_for_test in all failing test (all test that use the GGP) ?